### PR TITLE
Fixes unit parse error kind

### PR DESCRIPTION
**Pull request by m0n0chr0m3**
# Description

This PR fixes (what appear to be) two small copy/paste errors: on failure to parse from `&str`, both `Kilograms` and `Pounds` return a `ParseDistanceError` where presumably a `ParseMassError` should be returned.

I also implemented unit tests which would have revealed these problems in [a commit on a separate branch](https://github.com/m0n0chr0m3/fitnesstrax/commit/4f8b5e92774dedb0dfbc6883d9424867b4fa8a69). I did not include these updated unit tests in this PR, as I think the cleanest implementation makes use of the `matches!` macro or the `assert_matches!` macro. Prior to Rust 1.42, they require an external dependency: the `matches` crate. In Rust 1.42 the `matches!` macro [should land](https://github.com/rust-lang/rust/pull/67659) in the `std` lib.

In short, I think that the code in this PR can be accepted/rejected on short notice, and the followup commit can be kept around until you decide whether it's worth taking on another (dev-)dependency.

# Stories

This PR is not related to an existing issue or story. If the fate of the followup commit isn't decided here, I'll open an issue for it.

### CODE REVIEW

**Comment by savannidgerinel** [comment]
Good catch. Thank you.
